lucid/launcher.py

```diff
@@ -0,0 +1,116 @@
+def get_happi_entry_value(entry, key, search_extraneous=True):
+    extraneous = entry.extraneous
+    value = getattr(entry, key, None)
+    if value is None and search_extraneous:
+        # Try to look at extraneous
+        value = extraneous.get(key, None)
+
+    if not value:
+        raise ValueError('Invalid Key for Device.')
+    return value
+
+
+def main():
+    import logging
+    import argparse
+    from qtpy.QtWidgets import QApplication
+    import happi
+    import typhon
+    from . import __version__
+    from .main_window import LucidMainWindow
+    from .overview import IndicatorGrid
+
+    proj_desc = "LUCID - LCLS User Control and Interface Design"
+    parser = argparse.ArgumentParser(description=proj_desc)
+    parser.add_argument('--version', action='version',
+                        version='LUCID {version}'.format(version=__version__),
+                        help="Show LUCID's version number and exit.")
+
+    parser.add_argument(
+        'beamline',
+        help='Specify the beamline name to compose the home screen.',
+        type=str
+    )
+    parser.add_argument(
+        '--toolbar',
+        help='Path to the YAML file describing the entries for the Quick' +
+             ' Access Toolbar.',
+        default=None,
+        required=False,
+        type=argparse.FileType('r', encoding='UTF-8')
+    )
+    parser.add_argument(
+        '--row_group_key',
+        help='The Happi field to use for row grouping.',
+        default='location_group',
+        required=False
+    )
+    parser.add_argument(
+        '--col_group_key',
+        help='The Happi field to use for column grouping.',
+        default='functional_group',
+        required=False
+    )
+    parser.add_argument(
+        '--log_level',
+        help='Configure level of log display',
+        choices=['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL'],
+        default='INFO'
+    )
+
+    lucid_args = parser.parse_args()
```

Comment:
I really like the pattern of making the arg parsing separate from the actual functionality, such that you can programmatically use the launcher without having to make up your own list of string arguments.
e.g.,
```
    args = top_parser.parse_args()
    kwargs = vars(args)
    launcher(**kwargs)
```

where `kwargs` are `{'version', 'beamline', 'toolbar', ...}`. Though you might consider popping `version` out first and handling it specially.

Reply:
I like this idea! It sounds really good! The version I think will be handled by the parse_args special version case, right?

Reply:
I just learned about that option now, thanks!